packages/block-editor/src/store/private-selectors.js

```diff
@@ -282,11 +286,8 @@ export const hasAllowedPatterns = createRegistrySelector( ( select ) =>
 			} );
 		},
 		( state, rootClientId ) => [
-			getAllPatternsDependants( select )( state ),
-			state.settings.allowedBlockTypes,
-			state.settings.templateLock,
-			state.blockListSettings[ rootClientId ],
-			state.blocks.byClientId.get( rootClientId ),
+			...getAllPatternsDependants( select )( state ),
```

Comment:
This fixes a pre-existing bug: the array returned by `getAllPatternsDependants` must be spread out into the result. Otherwise the dependant is an array that's different on each call of `getAllPatternsDependants`, and the `hasAllowedPatterns` selector is not memoized at all.